### PR TITLE
[SHARE-465][Feature] Add endpoint for model types

### PR DESCRIPTION
**Pull request by leb2dg**
## Purpose

Provide a simple way to get all of the creative work types.

## Changes

Add `modeltypes` endpoint that returns the contents of `creative.yaml`.

## Side effects

None

### CODE REVIEW

File: api/views/share.py [hunk 1]
```diff
@@ -127,3 +130,11 @@ def get(self, request, *args, **kwargs):
                 return HttpSmartResponsePermanentRedirect(url)
             return HttpSmartResponseRedirect(url)
         return http.HttpResponseGone()
+
+
+class ModelTypesView(views.APIView):
+    def get(self, request, *args, **kwargs):
+        yaml_file = os.path.abspath('../SHARE/share/models/creative.yaml')
+        with open(yaml_file) as fobj:
+            model_specs = yaml.load(fobj)
```
Comment (chrisseto) [inline api/views/share.py]: This should be moved into the class definition so that it will only load once, on start up.

File: api/views/share.py [hunk 2]
```diff
@@ -127,3 +130,11 @@ def get(self, request, *args, **kwargs):
                 return HttpSmartResponsePermanentRedirect(url)
             return HttpSmartResponseRedirect(url)
         return http.HttpResponseGone()
+
+
+class ModelTypesView(views.APIView):
+    def get(self, request, *args, **kwargs):
+        yaml_file = os.path.abspath('../SHARE/share/models/creative.yaml')
```
Comment (chrisseto) [inline api/views/share.py]: This path will not work on all machines. Use the module level variable `__file__` to build the path.

File: api/views/share.py [hunk 3]
```diff
@@ -127,3 +130,11 @@ def get(self, request, *args, **kwargs):
...
+        with open(yaml_file) as fobj:
+            model_specs = yaml.load(fobj)
+        return Response(model_specs)
```
Comment (chrisseto) [inline api/views/share.py]: Will this be sent back in JSON API format?

File: api/urls.py [hunk 4]
```diff
@@ -94,5 +94,6 @@ def register_url(self, subclass, viewset):
     url(r'search/(?!.*_bulk\/?$)(?P<url_bits>.*)', csrf_exempt(views.ElasticSearchView.as_view()), name='search'),
 
     url(r'schema/?$', views.SchemaView.as_view(), name='schema'),
+    url(r'modeltypes/?$', views.ModelTypesView.as_view(), name='modeltypes'),
```
Comment (chrisseto) [inline api/urls.py]: This should be somewhere a bit more meaningful. Maybe just a generic `hierarchy/<modelname>` or `schema/<modelname>/types|hierarchy`?

File: api/views/share.py (hunk 3 above)
Comment (leb2dg) [inline api/views/share.py]: Yup!

File: api/urls.py (hunk 4 above)
Comment (leb2dg) [inline api/urls.py]: @chrisseto, are you suggesting that I redesign this entirely to support the hierarchy of all models?

Comment (aaxelb) [inline api/urls.py]: If you do, that'd be cool 👍

Comment (chrisseto) [inline api/urls.py]: @leb2dg Not really, that url pattern leaves room to do so if we'd like to.

File: api/views/schema.py [hunk 5]
```diff
@@ -113,3 +116,24 @@ def get(self, request, *args, **kwargs):
...
     }))
+
+
+class ModelTypesView(views.APIView):
+
+    def __init__(self, *args, **kwargs):
+        super(ModelTypesView, self).__init__(*args, **kwargs)
+        yaml_file = os.path.dirname(os.path.dirname(os.path.dirname(os.path.abspath(__file__)))) + '/share/models/creative.yaml'
+        with open(yaml_file) as fobj:
+            self.model_specs = self.sort_dict(yaml.load(fobj))
```
Comment (chrisseto) [inline api/views/schema.py]: This should be done at the class level.

File: api/views/schema.py [hunk 6]
```diff
@@ -113,3 +116,24 @@ def get(self, request, *args, **kwargs):
         'MODEL': model,
         '__doc__': ModelSchemaView.__doc__.format(model.__name__, model.__doc__)
     }))
+
+
+class ModelTypesView(views.APIView):
+
+    def __init__(self, *args, **kwargs):
+        super(ModelTypesView, self).__init__(*args, **kwargs)
+        yaml_file = os.path.dirname(os.path.dirname(os.path.dirname(os.path.abspath(__file__)))) + '/share/models/creative.yaml'
```
Comment (chrisseto) [inline api/views/schema.py]: Use `os.path.join` over the `+` here. Rather than using multiple `os.path.dirname`s throw `../` into the path.

**Pull request merged**